docs/developer/non-admin-controller.md

```diff
@@ -0,0 +1,48 @@
+# OADP NonAdminBackup Controller
+
+The OADP NonAdminBackup Kubernetes controller manages the lifecycle of custom resources called `NonAdminBackup`. Let's break down the code and understand its functionality.
+
+## Purpose
+
+The primary goal of this controller is to allow users with limited permissions (non-admins) to trigger and manage Velero backups within their namespaces. It acts as an intermediary between the non-admin user and Velero, ensuring that backups are created and managed securely within the defined boundaries.
+
+## Key Components
+
+### NonAdminBackup Custom Resource
+
+* Represents a backup request initiated by a non-admin user.
+* Contains a `BackupSpec` field, which mirrors the Velero `BackupSpec` and defines what to back up.
+* Tracks the status of the backup process through the `Status` field.
+
+### NonAdminBackupReconciler
+
+* The core controller responsible for processing `NonAdminBackup` resources.
+* Watches for changes to `NonAdminBackup` objects and triggers reconciliation loops.
+
+### Reconciliation Loop
+
+* **Init:** Initializes the `NonAdminBackup` status if it's a new object.
+* **ValidateSpec:** Ensures the provided `BackupSpec` is valid and adheres to security constraints.
+    * Rejects backups targeting namespaces outside the user's allowed scope.
+    * Sets appropriate conditions on the `NonAdminBackup` object to indicate success or failure.
+* **UpdateSpecStatus:**
```

Comment:
Based on the comment from @shubham-pampattiwar this function will probably be changed to some other name.